### ISSUE TITLE
api doesn't load component on production

### ISSUE THREAD

**Issue by octavioamu** [jellydn/next-swagger-doc#622]
### Describe the bug

Been trying option 2 (api /doc) following the docs and example and couldn't make it work and noticed on the demo is also broken https://next-swagger-doc-demo.productsway.com/api/doc schemas never get append to the json.

There is a way to make it append the schema as working on local? 
Even with yarn build yarn start on local works fine only failing on vercel

Note: I saw this been reported a few times but without to much information or solution.

### Reproduction

https://next-swagger-doc-demo.productsway.com/api/doc

### System Info

```Shell
System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 8.35 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
  Browsers:
    Brave Browser: 104.1.42.88
    Chrome: 109.0.5414.119
    Firefox: 106.0.2
    Safari: 16.2
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by jellydn** [comment]
Thanks, @octavioamu it does work on the demo https://next-swagger-doc-demo.productsway.com/api-doc. I will check tonight or so for the API endpoint.

**Comment by octavioamu** [comment]
@jellydn Thanks for responding.
That is because the api-doc is using the static generated file swagger.json which is the 3rd option. 
Same with the playground for example if you where using the 2nd to generate an api/doc probably because you want 3rd parties UI with the schemas you would use it like:
```
<APIDocumentUI
          apiDescriptionUrl="/api/doc"
          layout="sidebar"
          router="hash"
/>
```
 Instead of the static one https://github.com/jellydn/next-swagger-doc/blob/main/examples/example/pages/playground.tsx#L51
 
A good way to test it is looking the api/doc endpoint I posted https://next-swagger-doc-demo.productsway.com/api/doc
 when you run locally you will see the data will be in place when going localhost:3000/api/doc, even if you run `yarn build` and then `yarn start` api/doc will have the components schema there, which is really weird that got removed when deployed on vercel. 
 
 Spent my day yesterday testing different approach to try to see if I could make the api/doc append the schemas as local without good luck :( .

**Comment by octavioamu** [comment]
Also because is working on local with "yarn build" and not in vercel (that runs by default the same command) which is very strange my high guess is have something todo with async code on the server taking more time than local and for that not being appended.  But again, really high guess.

**Comment by octavioamu** [comment]
I think I know what is going on, you are using process.cwd() https://github.com/jellydn/next-swagger-doc/blob/main/src/swagger.ts#L39 I added a console log on the /api/doc endpoint and this is what cwd is on vercel as function are actually edge functions `/var/task` so the problem is how is being routed to find paths, the public folder is already "/" so probably doesn't need pwd to find the path and could go relative instead of absolute

**Comment by octavioamu** [comment]
I keep digging on this, one little suggestion is change join() to resolve() at least seems will be more secure if people input '/models/' on schemaFolders

**Comment by octavioamu** [comment]
no, resolve doesn't solve it, this is the server log of urls getting generated with resolve
```
[GET] /api/doc
15:55:31:07
process.cwd() 1 /var/task
apiss [
  '/var/task/pages/api/**/*.ts',
  '/var/task/pages/api/**/*.tsx',
  '/var/task/pages/api/**/*.jsx',
  '/var/task/pages/api/**/*.js',
  '/var/task/pages/api/**/*.json',
  '/var/task/pages/api/**/*.swagger.yaml',
  '/var/task/.next/server/pages/api/**/*.js',
  '/var/task/.next/server/pages/api/**/*.swagger.yaml',
  '/var/task/.next/server/pages/api/**/*.json',
  '/var/task/public/**/*.swagger.yaml',
  '/var/task/public/**/*.json',
  '/types/**/*.ts',
  '/types/**/*.tsx',
  '/types/**/*.jsx',
  '/types/**/*.js',
  '/types/**/*.json',
  '/types/**/*.swagger.yaml',
  '/types/**/*.js',
  '/types/**/*.swagger.yaml',
  '/types/**/*.json',
  '/var/task/public/**/*.swagger.yaml',
  '/var/task/public/**/*.json'
]
```

**Comment by jellydn** [comment]
I think we could have possible solutions here: 

- 1st solution: as far I as know, NextJs will remove comment on build project (`next build`) so we need to find a way to keep the comment code for generate swagger spec
- 2nd solution: write a custom Webpack plugin to generate swagger file on build

**Comment by octavioamu** [comment]
Interesting I tried with vercel-cli to avoid deploying each time, and worked, but now that Im thinking I didn't remove the folders created by yarn dev, so maybe on my local is working because still fetching from the other folders previously created.

**Comment by jellydn** [comment]
Hi @octavioamu I've just released version from #639 Could you take a try again? Thanks.

https://github.com/jellydn/next-swagger-doc/commit/01068361ae7649baa085a8c49dc616ad0fb346c8
